src/Shared/FileUtilities.cs

```diff
@@ -315,9 +315,14 @@ private static bool IsValidRelativePathBound(char? c)
         /// </summary>
         internal static string NormalizePath(string path)
         {
-
             ErrorUtilities.VerifyThrowArgumentLength(path, nameof(path));
+
+#if CLR2COMPATIBILITY
             string fullPath = GetFullPath(path);
+#else
+            // Windows uses a case-insensitive file system, so we need to get an exact file name from a possibly-differently-cased path.
+            string fullPath = NativeMethodsShared.IsWindows ? GetExactPath(path) : GetFullPath(path);
```

Comment:
It's not just windows. APFS and HFS+ are case-insensitive by default, and Linux supports some case-insensitive filesystems (though I don't think we have to care about that too much).

Reply:
Know of any way to detect whether the current FS is case sensitive or not? If there's no way currently, then doing it on windows-only is at least a good start.